source/javascripts/sliders2.js

```diff
@@ -158,6 +158,67 @@ tableApp2.controller('TableCtrl', function ($scope) {
 
     $scope.allLanguages = [
         {
+            name: "Safe Rust",
+            softname: "Up-and-coming systems lang",
+            markupName: "safe-rust",
+            comment: "//",
+
+            missingEnum: {
+                enforced: "yes",
+                desc: "Compiler enforced."
+            },
+            recursionStackOverflow: {
+                enforced: "no",
+                desc: "LLVM can compile tail calls in optimized mode, but this is not guaranteed, and, in fact, may silently go away if the compiler inserts a destructor call."
+            },
+            nullList: {
+                enforced: "yes",
+                desc: "Null pointers are never allowed, and Option must be explicitly handled."
+            },
+            deadLocks: {
+                enforced: "no",
+                desc: "Deadlocks are not considered unsafe."
+            },
+            missingMethodOrField: {
+                enforced: "yes",
+                desc: "Compiler Enforced."
+            },
+            wrongVaribleType: {
```

Comment:
Yeah, a few other languages allow this too, first to mind is F#. I didn't really think shadowing would be a source of errors, since the compilers in those languages are all going to correctly keep track of the which one refers to which type.